packages/vc-status-list/src/types/index.ts

```diff
@@ -23,33 +23,58 @@ import {
 } from '@veramo/core'
 import { DataSource } from 'typeorm'
 
-export interface CreateNewStatusListFuncArgs extends Omit<StatusList2021ToVerifiableCredentialArgs, 'encodedList'> {
-  correlationId: string
-  length?: number
+export type StatusList2021Args = {
+  encodedList: string
+  indexingDirection: StatusListIndexingDirection
+  statusPurpose?: StatusPurpose2021
+  // todo: validFrom and validUntil
 }
 
-export interface UpdateStatusListFromEncodedListArgs extends StatusList2021ToVerifiableCredentialArgs {
-  statusListIndex: number | string
-  value: boolean
+export type OAuth2StatusListArgs = {
+  status: 'active' | 'suspended' | 'revoked'
```

Comment:
Make this more in line with the StatusList2021Args. So statusPurpose and extract the type

Reply:
fixed